packages/monaco/src/browser/textmate/monaco-textmate-service.ts

```diff
@@ -23,13 +23,18 @@ import { LanguageGrammarDefinitionContribution, getEncodedLanguageId } from './t
 import { createTextmateTokenizer, TokenizerOption } from './textmate-tokenizer';
 import { TextmateRegistry } from './textmate-registry';
 import { MonacoThemeRegistry } from './monaco-theme-registry';
+import { EditorPreferences } from '@theia/editor/lib/browser/editor-preferences';
 
 export const OnigasmPromise = Symbol('OnigasmPromise');
 export type OnigasmPromise = Promise<IOnigLib>;
 
 @injectable()
 export class MonacoTextmateService implements FrontendApplicationContribution {
 
+    protected readonly tokenizerOption: TokenizerOption = {
+        lineLimit: 20_000
```

Comment:
IIRC that's way too much for Theia running in browser and led to short freezes. is it different now? did you try with bundle.js like files?

Reply:
👉 https://github.com/eclipse-theia/theia/pull/7618#issuecomment-616563467

Reply:
I cannot open production bundle.js at all in Theia :)

@AlexTugarev what do you think would be the reasonable length? 2000 sounds a lot too?

Reply:
I've tried with source map file with one line of 27408 characters and i don't have any issues:
<img width="1904" alt="Screenshot 2020-04-20 at 16 03 12" src="https://user-images.githubusercontent.com/3082655/79760542-8615f700-8320-11ea-8e04-ceabdf8c37d5.png">

Reply:
we measured once and ended up with 400

Reply:
@AlexTugarev Do you remember which files we used to test https://github.com/eclipse-theia/theia/issues/2599#issuecomment-414624168? I cannot reproduce these numbers: https://github.com/eclipse-theia/theia/issues/2599#issuecomment-414621596 We used latest vscode-textmate and onigasm now, maybe perf issues were resolved?

Reply:
ok, I will set to 400, since users can change it now. But we should now how to measure it to double check our assumptions.